### ISSUE TITLE
PercolatorAdapter crashes with concantenated Fractions 

### ISSUE THREAD

**Issue by aalvarezf** [OpenMS/OpenMS#4247]
Hi,
I'm analysing a TMT10p experiment with several fractions. 
I used FileMerger to concatenate the different mzML from 6 fractions in one file, with a gap in the RT. After that, the workflow is MSGFplus + PeptideIndexer + PSMFeatureExtractor + PercolatorAdapter 
The job crashes at the Percolator step with the error code below.
When I run the same workflow  with each fraction.mzML file, I got the results without any error. 

Could you help me to understand what is happening?
I repeat this and the error is always at the 3th split. 

Thanks!!

### ERROR CODE 
Executing: OpenMS/OpenMS/THIRDPARTY/Linux/64bit/Percolator/percolator
FATAL ERROR: External invocation of OpenMS/OpenMS/THIRDPARTY/Linux/64bit/Percolator/percolator failed. Standard output and error were:
Standard output: 
Standard error: Percolator version 3.02.0, Build Date Aug  5 2018 13:36:43
Copyright (c) 2006-9 University of Washington. All rights reserved.
Written by Lukas Käll (lukall@u.washington.edu) in the
Department of Genome Sciences at the University of Washington.
Issued command:
percolator -U -m _24958_2_target_pout_psms.tab -M _decoy_pout_psms.tab -Y _24958_2_pin.tab
Hyperparameters: selectionFdr=0.01, Cpos=0, Cneg=0, maxNiter=10
Reading tab-delimited input from datafile (...)24958_2_pin.tab
Features:
mass peplen charge2 charge3 charge4 enzN enzC enzInt dm absdm MS:1002049 MS:1002050 MSGF:ScoreRatio MSGF:Energy MSGF:lnEValue IsotopeError MSGF:lnExplainedIonCurrentRatio MSGF:lnNTermIonCurrentRatio MSGF:lnCTermIonCurrentRatio MSGF:lnMS2IonCurrent MSGF:MeanErrorTop7 MSGF:sqMeanErrorTop7 MSGF:StdevErrorTop7 
Found 388983 PSMs
Separate target and decoy search inputs detected, using target-decoy competition on Percolator scores.
Train/test set contains 220501 positives and 168482 negatives, size ratio=1.30875 and pi0=1
Selecting Cpos by cross-validation.
Selecting Cneg by cross-validation.
Split 1:	Selected feature 15 as initial direction. Could separate 27020 training set positives with q<0.01 in that direction.
Split 2:	Selected feature 15 as initial direction. Could separate 27020 training set positives with q<0.01 in that direction.
Split 3:	Exception caught: Error in the input data: cannot find an initial direction with positive training examples. Consider setting/raising the initial training FDR threshold (--train-initial-fdr).
Terminating.
Exit code: 1
PercolatorAdapter took 01:34 m (wall), 01:26 m (CPU), 0.87 s (system), 01:25 m (user).

**Comment by timosachsenberg** [comment]
One my data with difficult to separate scores I sometimes have a similar issue.
Sometimes I got it to work by increasing -trainFDR to e.g. 0.1

**Comment by aalvarezf** [comment]
Hi, and thanks for the answer.

I didn't find a proper answer for that behaviour of the PercolatorAdapter...  I post this on case it's worthy for someone

First, as you suggest (I think is also posted in other issues), I increase the -trainFDR and the testFDR, but finally I discard that option (should be increase a lot ).

I moved to the MSGFplus .jar and the Percolator binary with the exactly same parameters and the same  concatenated mzML, then converted the psms (target +decoy) to an idXML with IDFileConverter and continued with my workflow for protein inference via Fido, and as far as I know, It works.

Maybe I did something wrong with PercolatorAdapter, however I've found several issues using this tool, and this is the best solution I found. It looks quite strange to me, that the exactly same workflow works for each run but not for the concatenated version. I was worried about the data itself (maybe the svm does not discriminate properly all the psms,...) , but since  the concatenated version works with the binaries using the same parameters, I think that should be something  else.

I lost some info after the IDFileConvert, regarding MSGFp parameters, scan number etc, but since I do not need them in the next steps I move on with this solution.
